### PR TITLE
Enable conv/conv_transpose for opset 11 in cuda execution provider.

### PR DESCRIPTION
**Pull request by zhanghuanrong**

### CODE REVIEW

File: onnxruntime/core/providers/cuda/nn/conv.cc [hunk 1]
```diff
@@ -10,10 +10,18 @@ namespace onnxruntime {
...
 #define REGISTER_KERNEL_TYPED(T)                                                \
+  ONNX_OPERATOR_VERSIONED_TYPED_KERNEL_EX(                                      \
+      Conv,                                                                     \
+      kOnnxDomain,                                                              \
+      1, 10,                                                                    \
+      T,                                                                        \
+      kCudaExecutionProvider,                                                   \
+      KernelDefBuilder().TypeConstraint("T", DataTypeImpl::GetTensorType<T>()), \
+      Conv<T>);                                                                 \
   ONNX_OPERATOR_TYPED_KERNEL_EX(                                                \
```
Comment (skottmckay) [inline onnxruntime/core/providers/cuda/nn/conv.cc]: Please add a comment explaining why the opset 11 implementation can use the same code

**Comment by skottmckay** [comment]
Would it make sense to also add opset 11 versions for the pooling ops, as they were updated as part of the same change to onnx (https://github.com/onnx/onnx/pull/2291)?

**Comment by zhanghuanrong** [comment]
> Would it make sense to also add opset 11 versions for the pooling ops, as they were updated as part of the same change to onnx ([onnx/onnx#2291](https://github.com/onnx/onnx/pull/2291))?

Cool, Will add pool related changes here.

**Comment by zhanghuanrong** [comment]
Need your approval as merged conflict from the github UI.

File: onnxruntime/core/providers/cuda/cuda_execution_provider.cc [hunk 2]
```diff
@@ -635,6 +635,18 @@ class ONNX_OPERATOR_KERNEL_CLASS_NAME(kCudaExecutionProvider, kOnnxDomain, 11, S
...
+class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCudaExecutionProvider, kOnnxDomain, 11, MLFloat16, Conv);
+class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCudaExecutionProvider, kOnnxDomain, 11, float, ConvTranspose);
+class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCudaExecutionProvider, kOnnxDomain, 11, double, ConvTranspose);
+class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCudaExecutionProvider, kOnnxDomain, 11, MLFloat16, ConvTranspose);
+class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCudaExecutionProvider, kOnnxDomain, 11, float, AveragePool);
+class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCudaExecutionProvider, kOnnxDomain, 11, double, AveragePool);
+class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCudaExecutionProvider, kOnnxDomain, 11, MLFloat16, AveragePool);
+class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCudaExecutionProvider, kOnnxDomain, 11, float, MaxPool);
+class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCudaExecutionProvider, kOnnxDomain, 11, double, MaxPool);
+class ONNX_OPERATOR_TYPED_KERNEL_CLASS_NAME(kCudaExecutionProvider, kOnnxDomain, 11, MLFloat16, MaxPool);
```
Comment (skottmckay) [inline onnxruntime/core/providers/cuda/cuda_execution_provider.cc]: nit: all the other entries for opset 11 are sorted by name. would be nice to maintain that

File: onnxruntime/core/providers/cuda/nn/conv.cc [hunk 3]
```diff
@@ -9,11 +9,21 @@
 namespace onnxruntime {
 namespace cuda {
 
+// Op Set 11 for Conv only update document to clearify default dilations and strides value.
+// which are already convered by op set 11 cpu versoin, so simply add declaration.
```
Comment (skottmckay) [inline onnxruntime/core/providers/cuda/nn/conv.cc]: nit: clearify -> clarify

**Pull request merged**